src/Microsoft.TestPlatform.Extensions.BlameDataCollector/MiniDumpWriteDump.cs

```diff
@@ -0,0 +1,121 @@
+
+namespace Microsoft.TestPlatform.Extensions.BlameDataCollector;
+
+internal enum MiniDumpTypeOption
+{
+    // This is a copy of DumpTypeOption, we need both of those types, because this
+    // dumper is included as file in both MiniDumpTool, and BlameDataCollector, and
+    // blame references MiniDumpTool, so we get the enum defined twice otherwise.
+    Full,
+    WithHeap,
+    Mini,
+}
+
+internal class MiniDumpWriteDump
+{
+    public static void CollectDumpUsingMiniDumpWriteDump(Process process, string outputFile, MiniDumpTypeOption type)
+    {
+        // Open the file for writing
+        using var stream = new FileStream(outputFile, FileMode.Create, FileAccess.ReadWrite, FileShare.None);
+        NativeMethods.MinidumpExceptionInformation exceptionInfo = default;
+
+        NativeMethods.MinidumpType dumpType = NativeMethods.MinidumpType.MiniDumpNormal;
+        switch (type)
+        {
+            case MiniDumpTypeOption.Full:
+                dumpType = NativeMethods.MinidumpType.MiniDumpWithFullMemory |
+                           NativeMethods.MinidumpType.MiniDumpWithDataSegs |
+                           NativeMethods.MinidumpType.MiniDumpWithHandleData |
+                           NativeMethods.MinidumpType.MiniDumpWithUnloadedModules |
+                           NativeMethods.MinidumpType.MiniDumpWithFullMemoryInfo |
+                           NativeMethods.MinidumpType.MiniDumpWithThreadInfo |
+                           NativeMethods.MinidumpType.MiniDumpWithTokenInformation;
+                break;
+            case MiniDumpTypeOption.WithHeap:
+                dumpType = NativeMethods.MinidumpType.MiniDumpWithPrivateReadWriteMemory |
+                           NativeMethods.MinidumpType.MiniDumpWithDataSegs |
+                           NativeMethods.MinidumpType.MiniDumpWithHandleData |
+                           NativeMethods.MinidumpType.MiniDumpWithUnloadedModules |
+                           NativeMethods.MinidumpType.MiniDumpWithFullMemoryInfo |
+                           NativeMethods.MinidumpType.MiniDumpWithThreadInfo |
+                           NativeMethods.MinidumpType.MiniDumpWithTokenInformation;
+                break;
+            case MiniDumpTypeOption.Mini:
+                dumpType = NativeMethods.MinidumpType.MiniDumpWithThreadInfo;
+                break;
+        }
+
+        // Retry the write dump on ERROR_PARTIAL_COPY
+        for (int i = 0; i < 5; i++)
+        {
+            // Dump the process!
+            if (NativeMethods.MiniDumpWriteDump(process.Handle, (uint)process.Id, stream.SafeFileHandle, dumpType, ref exceptionInfo, IntPtr.Zero, IntPtr.Zero))
+            {
+                break;
+            }
+            else
+            {
+                int err = Marshal.GetHRForLastWin32Error();
+                if (err != NativeMethods.ErrorPartialCopy)
+                {
+                    Marshal.ThrowExceptionForHR(err);
+                }
+            }
```

Comment:
[cosmetic] Else is not required as you did early exiting.

```suggestion
            if (NativeMethods.MiniDumpWriteDump(process.Handle, (uint)process.Id, stream.SafeFileHandle, dumpType, ref exceptionInfo, IntPtr.Zero, IntPtr.Zero))
            {
                break;
            }
            
            int err = Marshal.GetHRForLastWin32Error();
            if (err != NativeMethods.ErrorPartialCopy)
            {
                Marshal.ThrowExceptionForHR(err);
            }
```

Reply:
done.